packages/edit-site/src/components/post-fields/index.js

```diff
@@ -395,8 +362,212 @@ function usePostFields( viewType ) {
+
+					const url = media?.source_url;
+					const title = media?.title?.rendered;
+
+					// This is a false positive
+					// eslint-disable-next-line react-hooks/rules-of-hooks
+					const { onClick } = useLink( {
+						postId: item.id,
+						postType: item.type,
+						canvas: 'edit',
+					} );
+
+					if ( viewType === LAYOUT_GRID && item.status !== 'trash' ) {
+						if ( ! url ) {
+							return null;
+						}
+						return (
+							<button
+								className="edit-site-post-list__featured-image-button"
+								type="button"
+								onClick={ onClick }
+								aria-label={
+									item.title?.rendered || __( '(no title)' )
+								}
+							>
+								<Media
+									className="edit-site-post-list__featured-image"
+									id={ item.featured_media }
+									size={ [
+										'large',
+										'full',
+										'medium',
+										'thumbnail',
+									] }
+								/>
+							</button>
+						);
+					}
+
+					if ( viewType === LAYOUT_LIST ) {
+						if ( ! url ) {
+							return null;
+						}
+
+						return (
+							<img
+								className="edit-site-post-featured-image"
+								src={ url }
+								alt=""
+							/>
+						);
+					}
+
+					if ( ! url ) {
+						return (
+							<Flex gap={ 8 }>
+								<FlexItem>
+									<span className="edit-site-post-featured-image-placeholder" />
+								</FlexItem>
+								<FlexItem>
+									<span>{ __( 'Choose an image…' ) }</span>
+								</FlexItem>
+							</Flex>
+						);
+					}
+
+					return (
+						<HStack>
+							<img
+								className="edit-site-post-featured-image"
+								src={ url }
+								alt=""
+							/>
+							<span>{ title }</span>
+						</HStack>
+					);
+				},
+				Edit: ( { field, onChange, data } ) => {
+					const { id } = field;
+
+					const value = field.getValue( { item: data } ) ?? '';
+
+					const onChangeControl = useCallback(
+						( newValue ) =>
+							onChange( {
+								[ id ]: newValue,
+							} ),
+						[ id, onChange ]
+					);
+
+					const media = getFeaturedMediaUrl( value );
+
+					const url = media?.source_url;
+					const title = media?.title?.rendered;
+					const filename =
+						media?.media_details?.file?.match( '([^/]+$)' )[ 0 ];
+
+					return (
+						<fieldset className="edit-site-dataviews-controls__featured-image">
+							<div className="edit-side-dataviews-controls__featured-image-container">
+								<MediaUpload
+									onSelect={ ( selectedMedia ) =>
+										onChangeControl( selectedMedia.id )
+									}
+									allowedTypes={ [ 'image' ] }
+									render={ ( { open } ) => {
+										return (
+											<div
+												role="button"
+												tabIndex={ 0 }
+												onClick={ ( event ) => {
+													const element =
+														event.target.tagName.toLowerCase();
+													// Prevent opening the media modal when clicking on the button/icon.
+													if (
+														element !== 'button' &&
+														element !== 'svg'
+													) {
```

Comment:
The issue is that there's a button within a clickable area, and clicking the button triggers a separate action. I'm not sure if this is the best way to handle it.

Reply:
Yeah we should avoid nested buttons. We had a similar issue in data views list layout... this UI:

<img width="384" alt="Screenshot 2024-08-23 at 11 41 02" src="https://github.com/user-attachments/assets/11cea8e7-515d-423e-a8fe-059d983f0311">

Perhaps some of the techniques there can be borrowed?